pkgs/tools/security/one-password-cli/default.nix

```diff
@@ -0,0 +1,72 @@
+{ lib
+, stdenv
+, autoPatchelfHook
+, fetchzip
+}:
+
+let
+  archNameMapping = {
+    i686 = "386";
+    x86_64 = "amd64";
+    aarch64 = "arm";
+    armv5tel = "arm";
+    armv6l = "arm";
+    armv7a = "arm";
+    armv7l = "arm";
+  };
+  archName = archNameMapping.${stdenv.targetPlatform.uname.processor};
+  kernelName = stdenv.targetPlatform.parsed.kernel.name;
+in
+
+stdenv.mkDerivation rec {
+  pname = "one-password";
+  version = "1.8.0";
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  src = fetchzip {
+    url = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_${kernelName}_${archName}_v${version}.zip";
+    sha256 = "1651ricswsdq9yi742k4403qar7cvfc2glr4wr2glga7f891spax";
```

Comment:
There are multiple URLs based on the architecture, but only one checksum? This should also be defined in a map.